gensim/models/experimental/drmm_tks.py

```diff
@@ -0,0 +1,840 @@
+                        if new_item[1] == 0:
+                            yield(_make_indexed(q), _make_indexed(item[0]), _make_indexed(new_item[0]))
+
+
+class DRMM_TKS(utils.SaveLoad):
+    """Model for training a Similarity Learning Model using the DRMM TKS model.
+    You only have to provide sentences in the data as a list of words.
+    """
+
+    def __init__(self, queries=None, docs=None, labels=None, word_embedding=None,
+                 text_maxlen=200, normalize_embeddings=True, epochs=10, unk_handle_method='random',
+                 validation_data=None, topk=50, target_mode='ranking', verbose=1):
+        """Initializes the model and trains it
+
+        Parameters
+        ----------
+        queries: iterable list of list of string words, optional
+            The questions for the similarity learning model.
+        docs: iterable list of list of list of string words, optional
+            The candidate answers for the similarity learning model.
+        labels: iterable list of list of int, optional
+            Indicates when a candidate document is relevant to a query
+            - 1 : relevant
+            - 0 : irrelevant
+        word_embedding : :class:`~gensim.models.keyedvectors.KeyedVectors`, optional
+            a KeyedVector object which has the embeddings pre-loaded.
+            If None, random word embeddings will be used.
+        text_maxlen : int, optional
+            The maximum possible length of a query or a document.
+            This is used for padding sentences.
+        normalize_embeddings : bool, optional
+            Whether the word embeddings provided should be normalized.
+        epochs : int, optional
+            The number of epochs for which the model should train on the data.
+        unk_handle_method : {'zero', 'random'}, optional
+            The method for handling unkown words.
+                - 'zero' : unknown words are given a zero vector
+                - 'random' : unknown words are given a uniformly random vector bassed on the word string hash
+        validation_data: list of the form [test_queries, test_docs, test_labels], optional
+            where test_queries, test_docs  and test_labels are of the same form as
+            their counter parts stated above.
+        topk : int, optional
+            the k topmost values in the interaction matrix between the queries and the docs
+        target_mode : {'ranking', 'classification'}, optional
+            the way the model should be trained, either to rank or classify
+        verbose : {0, 1, 2}
+            the level of information shared while training
+                - 0 : silent
+                - 1 : progress bar
+                - 2 : one line per epoch
+
+
+        Examples
+        --------
+        The trained model needs to be trained on data in the format
+
+        >>> queries = ["When was World War 1 fought ?".lower().split(), "When was Gandhi born ?".lower().split()]
```

Comment:
unfortunatelly unreadable example (all in a "heap"), follow my instruction in comment below (top of file)